include/LightGBM/utils/common.h

```diff
@@ -580,20 +580,26 @@ static void ParallelSort(_RanIt _First, _RanIt _Last, _Pr _Pred) {
 }
 
 // Check that all y[] are in interval [ymin, ymax] (end points included); throws error if not
-inline void CheckElementsIntervalClosed(const float *y, float ymin, float ymax, int ny, const char *callername) {
+template <typename T>
+inline void CheckElementsIntervalClosed(const T *y, T ymin, T ymax, int ny, const char *callername) {
   for (int i = 0; i < ny; ++i) {
     if (y[i] < ymin || y[i] > ymax) {
-      Log::Fatal("[%s]: does not tolerate element [#%i = %f] outside [%f, %f]", callername, i, y[i], ymin, ymax);
+      if (typeid(T) == typeid(float) || typeid(T) == typeid(double)) {
```

Comment:
why this ? for the format output ? 
you can use the stringstream to format ``T`` type.